config/m4/nl.m4

```diff
@@ -16,32 +16,40 @@ else
 	libibverbs_file="$ac_cv_ofed_path/lib/libibverbs.so"
 fi
 
-have_libnl1=no
-have_libnl3=no
+PKG_CHECK_MODULES([LIBNL3],[libnl-route-3.0],[use_libnl3=yes] ,[use_libnl3=no])
+PKG_CHECK_MODULES([LIBNL1],[libnl-1], [use_libnl1=yes] , [use_libnl1=no])
 
 ldd $libibverbs_file | grep libnl >/dev/null 2>&1
 if test $? -eq 0 ; then
+	# vma must use the smae libnl verbs uses
 	ldd $libibverbs_file | grep -e 'libnl3' -e 'libnl-3' >/dev/null 2>&1
 	if test $? -eq 0 ; then
 		# libnl3 case
-		PKG_CHECK_MODULES([LIBNL3],[libnl-route-3.0],have_libnl3=yes AC_DEFINE([HAVE_LIBNL3], [1], [Use libnl-route-3.0]),[:])
-		if test "$have_libnl3" == no; then
+		if test "$use_libnl3" == no; then
 			AC_MSG_ERROR([libibverbs is linked with libnl3 while libnl3-devel is not installed. Please install libnl3-devel and try again])
 		fi
-		AC_SUBST([LIBNL_LIBS], "$LIBNL3_LIBS")
-		AC_SUBST([LIBNL_CFLAGS], "$LIBNL3_CFLAGS")
-		AC_SUBST([LIBNLX_DEVEL], "libnl3-devel")
+		use_libnl1=no
 	else
 		# libnl1 case
-		PKG_CHECK_MODULES([LIBNL1],[libnl-1], have_libnl1=yes AC_DEFINE([HAVE_LIBNL1], [1], [Use libnl-1]), [:])
-		if test "$have_libnl1" == no; then
+		if test "$use_libnl1" == no; then
 			AC_MSG_ERROR([libibverbs is linked with libnl1 while libnl1-devel is not installed. Please install libnl1-devel and try again])
 		fi
-		AC_SUBST([LIBNL_LIBS], "$LIBNL1_LIBS")
-		AC_SUBST([LIBNL_CFLAGS], "$LIBNL1_CFLAGS")
-		AC_SUBST([LIBNLX_DEVEL], "libnl-devel")
+		use_libnl3=no
 	fi
 fi
 
-AM_CONDITIONAL([HAVE_LIBNL], [test "$have_libnl1" = "yes" -o "$have_libnl3" = "yes"])
+if test "$use_libnl3" == yes; then
+	AC_SUBST([LIBNL_LIBS], "$LIBNL3_LIBS")
+	AC_SUBST([LIBNL_CFLAGS], "$LIBNL3_CFLAGS")
+	AC_SUBST([LIBNLX_DEVEL], "libnl3-devel")
+	AC_DEFINE([HAVE_LIBNL3], [1], [Use libnl-route-3.0])
+elif test "$use_libnl1" == yes; then
+	AC_SUBST([LIBNL_LIBS], "$LIBNL1_LIBS")
+	AC_SUBST([LIBNL_CFLAGS], "$LIBNL1_CFLAGS")
+	AC_SUBST([LIBNLX_DEVEL], "libnl-devel")
+	AC_DEFINE([HAVE_LIBNL1], [1], [Use libnl-1])
+else
+	AC_MSG_ERROR([libibverbs needs libnl3-devel\libnl1-devel (better libnl3-devel)])
+fi
+
```

Comment:
What is about just following
```
if test "$have_libnl1" = "yes" -o "$have_libnl3" = "yes"; then
   AM_CONDITIONAL([HAVE_LIBNL], [1])
else
   AC_MSG_ERROR([libibverbs needs libnl3-devel\libnl1-devel (better libnl3-devel)])
fi
```
instead of
```
AM_CONDITIONAL([HAVE_LIBNL], [test "$have_libnl1" = "yes" -o "$have_libnl3" = "yes"])
```

Reply:
HAVE_LIBNL is not used